pkg/actuators/machine/instances.go

```diff
@@ -422,9 +427,28 @@ func buildTagList(machineName string, clusterID string, machineTags []awsprovide
 		{Key: aws.String("kubernetes.io/cluster/" + clusterID), Value: aws.String("owned")},
 		{Key: aws.String("Name"), Value: aws.String(machineName)},
 	}...)
+
 	return removeDuplicatedTags(rawTagList)
 }
 
+// mergeInfrastructureAndMachineSpecTags merge list of tags from machine provider spec and Infrastructure object platform spec.
+// Machine tags have precedence over Infrastructure
+func mergeInfrastructureAndMachineSpecTags(machineSpecTags []awsproviderv1.TagSpecification, infra *configv1.Infrastructure) []awsproviderv1.TagSpecification {
+	if infra == nil || infra.Status.PlatformStatus == nil || infra.Status.PlatformStatus.AWS == nil || infra.Status.PlatformStatus.AWS.ResourceTags == nil {
+		return machineSpecTags
+	}
+
+	mergedList := []awsproviderv1.TagSpecification{}
+
+	for _, tag := range infra.Status.PlatformStatus.AWS.ResourceTags {
+		mergedList = append(mergedList, awsproviderv1.TagSpecification{Name: tag.Key, Value: tag.Value})
+	}
+
+	machineSpecTags = append(machineSpecTags, mergedList...)
```

Comment:
This looks like we're allowing for duplicates in the list which is unsupported I believe.  Also, the enhancement specifies tags set on the machineset should override those in the infra.Status, so we need to build a map, first with the infra.Status tags, then apply the machineset, then dump to list.

Reply:
The tags returned from this list are deduplicated once the name and cluster tags are added, look at approx line 431 of this file. Based on the unit testing, I think this is all working as expected

Reply:
yes, tags are deduplicated later